src/contexts/FormElementUpdateTrackerState.tsx

```diff
@@ -35,11 +36,17 @@ const reducer = (
         elementUpdatedTimestamp: Date.now(),
         elementUpdatedTextValue: action.textValue,
       }
+
+      const notEntryState = typeof newState.elementEnteredTextValue === 'undefined'
+      let wasElementChanged = action.textValue !== newState.elementEnteredTextValue
+      if (notEntryState) wasElementChanged = action.textValue !== action.previousValue
```

Comment:
Can this not just be:
```
const wasElementChanged =
        typeof newState.elementEnteredTextValue === 'undefined'
          ? action.textValue !== action.previousValue
          : action.textValue !== newState.elementEnteredTextValue
```

Personally I find it easier to read, because the name "notEntryState" is kind of meaningless to me. So I have to parse its contents anyway, whereas the ternary structure at least formats it in a readable shape. Just a personal preference, no biggie.

Reply:
I agree: 3a241b6e76a272941409fa5faab75e5e1f0bd0da